pyccl/nonlimber_FKEM.py

```diff
@@ -0,0 +1,234 @@
+__all__ = ("nonlimber_FKEM",)
+"""Written by Paul Rogozenski (paulrogozenski@arizona.edu),
+ implementing the FKEM non-limber integration method of the N5K challenge
+ detailed in this paper: https://arxiv.org/pdf/1911.11947.pdf .
+We utilize a modified generalized version of FFTLog
+ (https://jila.colorado.edu/~ajsh/FFTLog/fftlog.pdf)
+ to compute integrals over spherical bessel functions
+"""
+import numpy as np
+from . import lib, check
+from .pyutils import integ_types
+from scipy.interpolate import interp1d
+from pyccl.pyutils import _fftlog_transform_general
+import pyccl as ccl
+
+
+def get_general_params(b):
+    nu = 1.51
+    nu2 = 0.51
+    deriv = 0.0
+    plaw = 0.0
+    best_nu = nu
+    if b < 0:
+        plaw = -2.0
+        best_nu = nu2
+
+    if b <= 0:
+        deriv = 0
+    else:
+        deriv = b
+
+    return best_nu, deriv, plaw
+
+
+def get_average_a(a_arr, dndz):
+    z_arr = 1.0 / (a_arr) - 1
+    dz = (z_arr[-1] - z_arr[0]) / (len(z_arr) - 1)
+    new_arr = []
+    for i in range(len(dndz)):
+        if dndz[i] != 0:
+            new_arr.append(dz * z_arr[i] * dndz[i])
+    z_mean = np.sum(np.array(new_arr))
+    return 1.0 / (1.0 + z_mean)
+
+
+def nonlimber_FKEM(
+    cosmo, clt1, clt2, p_of_k_a, ls, l_limber, limber_max_error
+):
```

Comment:
This would need proper documentation if it's meant to be used as part of the API. If it isn't, then it should be made private.